api/src/main/java/org/commonjava/indy/data/ArtifactStoreValidateData.java

```diff
@@ -0,0 +1,78 @@
+package org.commonjava.indy.data;
+
+import org.apache.kafka.common.protocol.types.Field;
+import org.bouncycastle.util.Store;
+import org.commonjava.indy.model.core.StoreType;
+
+public class ArtifactStoreValidateData {
+
+    private boolean valid;
+    private String repositoryUrl;
+    private StoreType storeType;
+
+    private ArtifactStoreValidateData() {
+    }
+
+
+    public static class Builder {
+
+        private boolean valid;
+        private String repositoryUrl;
+        private StoreType storeType;
+
+        public Builder(boolean valid) {
```

Comment:
**"I think it would also be good to have a Map<String, String> field containing validation errors. If we use a common set of keys for the different validation rules..."**

Is This good for our scenario , because we should add error messages on return object for validate method and this would break our concept for throwing checked exceptions and handle not valid repos in caller method. If i do not miss something ,  this scenario ( with returning object with error map ) will mean that we will need to return ArtifactStoreValidateData object ( with only that exception as value )  on each thrown exception and not to throw checked exception (MailformedURLException , InvalidArtifactStoreException ).
BTW i agree that nice option will be to have central place object where we could list validation exceptions for some store and display that data, but isn't it better if we gather that exception data outside DefaultArtifactStore.validation method , example from caller methods?